comet/debrid/stremthru.py

```diff
@@ -0,0 +1,178 @@
+import asyncio
+from typing import Optional
+
+import aiohttp
+from RTN import parse
+
+from comet.utils.general import is_video
+from comet.utils.logger import logger
+
+
+class StremThru:
+    def __init__(
+        self,
+        session: aiohttp.ClientSession,
+        url: str,
+        store: str,
+        store_token: Optional[str] = None,
+        credential: Optional[str] = None,
+    ):
+        if not credential and not store_token:
+            raise TypeError("either stremthru credential or store token is required")
+
+        if not self.is_supported_store(store):
+            raise ValueError(f"unsupported store: {store}")
+
+        self.__is_proxy_authorized = False
+
+        session.headers["X-StremThru-Store-Name"] = store
+
+        if credential:
+            session.headers["Proxy-Authorization"] = f"Basic {credential}"
+            self.__is_proxy_authorized = True
+        elif store_token:
+            session.headers["X-StremThru-Store-Authorization"] = f"Bearer {store_token}"
+
+        self.session = session
+        self.base_url = f"{url}/v0/store"
+        self.name = f"StremThru[{store}]"
+
+    def should_try_to_proxy_stream(self):
+        return not self.__is_proxy_authorized
```

Comment:
When proxy authorized, StremThru will do the proxy streaming.

Comet just needs to redirect to the generated link.